CineElorrietaMVC/src/main/java/Vista/PanelRegistro.java

```diff
@@ -0,0 +1,183 @@
+		lblDNI.setBounds(330, 79, 81, 30);
+		add(lblDNI);
+
+		JLabel lblContrasenna = new JLabel("Contrase\u00F1a:");
+		lblContrasenna.setFont(new Font("Arial", Font.PLAIN, 17));
+		lblContrasenna.setBounds(33, 259, 127, 23);
+		add(lblContrasenna);
+
+		JLabel lblNombre = new JLabel("Nombre:");
+		lblNombre.setFont(new Font("Arial", Font.PLAIN, 17));
+		lblNombre.setBounds(28, 82, 89, 24);
+		add(lblNombre);
+
+		JLabel lblApellido = new JLabel("Apellido:");
+		lblApellido.setFont(new Font("Arial", Font.PLAIN, 17));
+		lblApellido.setBounds(28, 117, 106, 20);
+		add(lblApellido);
+
+		textDNI = new JTextField();
+
+		textDNI.setBounds(371, 86, 141, 20);
+		add(textDNI);
+		textDNI.setColumns(10);
+
+		textNombre = new JTextField();
+
+		textNombre.setBounds(101, 86, 169, 20);
+		add(textNombre);
+		textNombre.setColumns(10);
+
+		textApellido = new JTextField();
+
+		textApellido.setBounds(101, 120, 169, 20);
+		add(textApellido);
+		textApellido.setColumns(10);
+
+		passwordContrasena = new JPasswordField();
+
+		passwordContrasena.setBounds(137, 262, 133, 20);
+		add(passwordContrasena);
+
+		initializeEvents();
+
+	}
+
+	private void initializeEvents() {
+		this.btnVolver.addActionListener(listenerBotonVolver(this.controlador));
+		this.btnRegistrar.addActionListener(listenerBotonRegistrar(this.controlador));
+	}
+
+	private ActionListener listenerBotonVolver(ControladorPanelRegistro controladorPanelRegistro) {
+		return new ActionListener() {
+			public void actionPerformed(ActionEvent arg0) {
+				System.out.println("Ejecutando evento Boton Volver");
+				controladorPanelRegistro.accionadoBottonVolverPanelLogin();
+			}
+		};
+	}
+
+	private ActionListener listenerBotonRegistrar(ControladorPanelRegistro controladorPanelRegistro) {
+		return new ActionListener() {
+			public void actionPerformed(ActionEvent arg0) {
+				System.out.println("Ejecutando evento Boton Registrar");
+
+				String nombre = textNombre.getText();
+				String dni = textDNI.getText();
+				String nif = textNIF.getText();
+				String apellido = textApellido.getText();
+				@SuppressWarnings("deprecation")
+				String password = passwordContrasena.getText();
+
+				if (controladorPanelRegistro.comprobarFormatoNombre(nombre)
+						&& controladorPanelRegistro.comprobarFormatoApellido(apellido)
+						&& controladorPanelRegistro.comprobarNif(nif) && controladorPanelRegistro.comprobarNif(dni)
+						&& controladorPanelRegistro.comprobarBBDDnif(nif)
+						&& !controladorPanelRegistro.comprobarBBDDdni(dni)
+						&& controladorPanelRegistro.comprobarContraNoVacia(password))
+
+				{
+					// insertar datos en empleado
+					controladorPanelRegistro.insertarRegistro(dni, nombre, apellido, password, nif);
+
+					JOptionPane.showMessageDialog(null, "Empleado introducido correctamente");
+					controladorPanelRegistro.accionadoBottonVolverPanelLogin();
+
+				} else {
+					if (controladorPanelRegistro.comprobarFormatoNombre(nombre) == false) {
```

Comment:
Lo mismo en la llamada de comprobar datos podéis devolver el mensaje de que campos no está correcto (incluso ir concatenandolos) y así os evitáis tanto if

Reply:
Corregido.

Reply:
Corregido.